### ISSUE TITLE
usleep is not available on Windows

### ISSUE THREAD

**Issue by KOLANICH** [dmlc/dmlc-core#456]
https://github.com/dmlc/dmlc-core/blob/649be18a8c55c48517861d67158a45dec54992ee/test/unittest/unittest_thread_group.cc#L32

`usleep` is used. It is not available (in fact, it is, see https://github.com/KOLANICH/nixShims4Windows/blob/master/src/time.c#L2 , it uses the undocumented function) on Windows.

You have an own shim, but it is enabled only for MSVC
https://github.com/dmlc/dmlc-core/blob/649be18a8c55c48517861d67158a45dec54992ee/test/unittest/unittest_thread_group.cc#L10

But it is not needed, 1e4 microseconds is 10 milliseconds, whic is in range of [`Sleep`](https://docs.microsoft.com/en-us/windows/desktop/api/synchapi/nf-synchapi-sleep)

**Comment by hcho3** [comment]
Changing `#if defined(_MSC_VER)` to `#if defined(_WIN32)` should solve the problem.

**Comment by KOLANICH** [comment]
Why not just use sleep requiring time in milliseconds?

**Comment by hcho3** [comment]
Isn't Sleep specific to Windows?

**Comment by hcho3** [comment]
Since dmlc-core has to support multiple OSses, we have no choice but to have shims for OS-specific functions, such as sleep functions.

**Comment by KOLANICH** [comment]
can we use `std::this_thread::sleep_for`?

**Comment by hcho3** [comment]
Yes, we can, but see our policy regarding the use of C++11 (from front page):

> Use of c++11 is allowed, given that the code is macro guarded with DMLC_USE_CXX11

So we'll still need a shim for users of C++98 / C++0x.

**Comment by hcho3** [comment]
#462 should fix this.
